contracts/RewardsManager.vy

```diff
@@ -54,33 +174,58 @@ def is_rewards_period_finished() -> bool:
     """
     @notice Whether the current rewards period has finished.
     """
-    return self._is_rewards_period_finished(self.rewards_contract)
+    return block.timestamp >= self._period_finish()
+    
+
+@external
+def transfer_ownership(_to: address):
+    """
+    @notice Changes the contract owner. Can only be called by the current owner.
+    """
+    current_owner: address = self.owner
+    assert msg.sender == current_owner, "not permitted"
+    self.owner = _to
+    log OwnershipTransferred(current_owner, _to)
 
 
-@view
 @external
-def period_finish() -> uint256:
+def transfer_rewards_contract(_to: address):
     """
-    @notice Returns end of the rewards period of StakingRewards contract
+    @notice Changes the reward contracts distributor. Can only be called by the current owner.
     """
-    return self._period_finish(self.rewards_contract)
+    assert msg.sender == self.owner, "not permitted"
+    assert _to != ZERO_ADDRESS, "zero address not allowed"
+    BalancerLiquidityGauge(self.rewards_contract).set_reward_distributor(LDO_TOKEN, _to)
+
+    log RewardsContractTransferred(_to)
+
 
 @external
-def start_next_rewards_period():
+def set_rewards_contract(_rewards_contract: address):
```

Comment:
Do we need to ensure that there is no ongoing reward distribution period before changing the contract?

Reply:
I think we don't, starting new period after previous finished is limitation of permissionless logic of current contract, we should be able to transfer distribution permission anyway if that will be required